### PR TITLE
Split out Istio testgrid config and use autogen

### PR DESCRIPTION
**Pull request by howardjohn**
We moved Istio to using annotations, so now our config is autogenerated.
This PR deletes all the big config.yaml and splits it into the istio
folder, where the new generated config is checked in.

### CODE REVIEW

**Comment by howardjohn** [comment]
/hold

let me wait until I have some time to fix this if it breaks our dashboards before merging

**Comment by chases2** [comment]
xref #13541 
/ok-to-test

File: config/testgrids/istio/istio.yaml [hunk 1]
```diff
@@ -0,0 +1,1222 @@
+default_dashboard_tab:
+  attach_bug_template: {}
+  code_search_path: github.com/istio/istio/search
```
Comment (chases2) [inline config/testgrids/istio/istio.yaml]: Default settings won't be respected (see #13636 )

The newest version of Configurator should regenerate these files without the defaults. Alternately, please delete lines 1-29

File: config/testgrids/config.yaml [hunk 2]
```diff
@@ -2585,559 +2071,6 @@ dashboards:
 - name: presubmits-kube-batch
 - name: presubmits-node-problem-detector
 - name: google-rules_k8s
-- name: istio-presubmits
```
Comment (chases2) [inline config/testgrids/config.yaml]: This dashboard is deleted, but not present in your configuration.

Was this intentional? Is `istio-presubmits` spelled correctly in the generating yaml doc?

File: config/testgrids/config.yaml [hunk 3]
```diff
@@ -2585,559 +2071,6 @@ dashboards:
...
-  - name: circleci-codecov-tests
-    test_group_name: istio-circleci-codecov-presubmit-release-1.2
-  - name: circleci-racetest
-    test_group_name: istio-circleci-racetest-presubmit-release-1.2
-  - name: circleci-integration-local
-    test_group_name: istio-circleci-test-integration-local-presubmit-release-1.2
-  - name: circleci-integration-kubernetes
-    test_group_name: istio-circleci-test-integration-kubernetes-presubmit-release-1.2
-
-- name: istio-postsubmits
```
Comment (chases2) [inline config/testgrids/config.yaml]: Same issue here; there's an `istio-postsubmits-release-1.2` and an `istio-postsubmits-master`, but not `istio-postsubmits`

File: config/testgrids/config.yaml (hunk 2 above)
Comment (howardjohn) [reply config/testgrids/config.yaml]: This is intentional, these are legacy from before we split tests by branch

**Comment by howardjohn** [comment]
@chases2 updated all the test grids in https://github.com/istio/test-infra/pull/1565 to include everything so tests are passing now. Thanks for the help!

**Pull request merged**